File: 프로그래머스/0/120860. 직사각형 넓이 구하기/직사각형 넓이 구하기.java
import java.util.*;
class Solution {
    public int solution(int[][] dots) {
        int answer = 0;

        int x_min = Integer.MAX_VALUE;
        int x_max = Integer.MIN_VALUE;
        int y_min = Integer.MAX_VALUE;
        int y_max = Integer.MIN_VALUE;

        for(int i = 0; i < dots.length; i++){
            if(dots[i][0] < x_min) x_min = dots[i][0];
            if(dots[i][0] > x_max) x_max = dots[i][0];
            if(dots[i][1] < y_min) y_min = dots[i][1];
            if(dots[i][1] > y_max) y_max = dots[i][1];
        }
        
        answer = Math.abs(x_max - x_min) * Math.abs(y_max - y_min);
        
        return answer;
    }
}

File: 프로그래머스/0/120860. 직사각형 넓이 구하기/직사각형 넓이 구하기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(int** dots, size_t dots_rows, size_t dots_cols) {
    int answer = 0;
    
    int x = max(abs(dots[0][0] - dots[1][0]), abs(dots[1][0] - dots[2][0]));
    int y = max(abs(dots[0][1] - dots[1][1]), abs(dots[1][1] - dots[2][1]));
        
    answer = x * y;
    
    return answer;
}

int max(int a, int b){
    return a > b ? a : b;
}